lib/Logger/Logger.cpp

```diff
@@ -704,64 +714,62 @@ void Logger::append(LogGroup& group,
 ////////////////////////////////////////////////////////////////////////////////
 
 void Logger::initialize(application_features::ApplicationServer& server, bool threaded) {
-  MUTEX_LOCKER(locker, _initializeMutex);
-
-  if (_active.exchange(true)) {
+  if (_active.exchange(true, std::memory_order_acquire)) {
     THROW_ARANGO_EXCEPTION_MESSAGE(TRI_ERROR_INTERNAL,
                                    "Logger already initialized");
   }
 
   // logging is now active
-  TRI_ASSERT(_active);
-  
   if (threaded) {
-    _loggingThread = std::make_unique<LogThread>(server, ::LogThreadName);
-    if (!_loggingThread->start()) {
+    auto loggingThread = new LogThread(server, ::LogThreadName);
+    if (!loggingThread->start()) {
       LOG_TOPIC("28bd9", FATAL, arangodb::Logger::FIXME)
           << "could not start logging thread";
       FATAL_ERROR_EXIT();
     }
-  }
   
-  // generate threaded logging?
-  _threaded = threaded;
+    _loggingThread.store(loggingThread, std::memory_order_release);
+  }
 }
 
 ////////////////////////////////////////////////////////////////////////////////
 /// @brief shuts down the logging components
 ////////////////////////////////////////////////////////////////////////////////
 
 void Logger::shutdown() {
-  MUTEX_LOCKER(locker, _initializeMutex);
-
-  if (!_active.exchange(false)) {
-    // if logging not activated or already shutdown, then we can abort here
+  if (!_active.exchange(false, std::memory_order_acquire)) {
+    // if logging not activated or already shut down, then we can abort here
     return;
   }
+  // logging is now inactive
 
-  TRI_ASSERT(!_active);
-
+  // reset the instance variable in Logger, so that others won't see it anymore
+  auto loggingThread = _loggingThread.exchange(nullptr, std::memory_order_relaxed);
```

Comment:
I think `loggingThread` will be a raw pointer here, so it will be leaked once the function ends. Can we capture in a unique_ptr, so that whenever we exit this function, we'll destroy it.
```suggestion
  auto loggingThread = std::unique_ptr<LogThread>(_loggingThread.exchange(nullptr, std::memory_order_relaxed));
```